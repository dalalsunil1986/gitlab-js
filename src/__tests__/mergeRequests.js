jest.mock('../utils/api', () => jest.fn());
jest.mock('extra-log', () => ({ msg: jest.fn() }));

const mergeRequest = require('../mergeRequest');
const api = require('../utils/api');

describe('request', () => {
  beforeEach(() => {
    api.mockReset();
    api.mockImplementation((...params) => [...params]);
  });


  it('list', async () => {
    const response = await mergeRequest.list('projectId');

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('list, no params', async () => {
    const response = await mergeRequest.list();

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(0);
  });


  it('get', async () => {
    const response = await mergeRequest.get('projectId', 'mergeRequestId');

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('get, no params', async () => {
    expect(await mergeRequest.get()).toMatchSnapshot();
    expect(await mergeRequest.get('projectId')).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(0);
  });


  it('notes', async () => {
    const response = await mergeRequest.notes('projectId', 'mergeRequestId');

    expect(response).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(1);
    expect(api.mock.calls).toMatchSnapshot();
  });


  it('notes, no params', async () => {
    expect(await mergeRequest.notes()).toMatchSnapshot();
    expect(await mergeRequest.notes('projectId')).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(0);
  });


  it('approve', async () => {
    expect(await mergeRequest.approve()).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(0);
  });


  it('accept', async () => {
    expect(await mergeRequest.approve()).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(0);
  });


  it('unaprove', async () => {
    expect(await mergeRequest.approve()).toMatchSnapshot();
    expect(api).toHaveBeenCalledTimes(0);
  });
});
