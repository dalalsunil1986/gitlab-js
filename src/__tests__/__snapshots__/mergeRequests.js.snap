// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request accept 1`] = `undefined`;

exports[`request approve 1`] = `undefined`;

exports[`request get 1`] = `
Array [
  "/projects/projectId/merge_requests/mergeRequestId",
  Object {
    "simple": true,
  },
]
`;

exports[`request get 2`] = `
Array [
  Array [
    "/projects/projectId/merge_requests/mergeRequestId",
    Object {
      "simple": true,
    },
  ],
]
`;

exports[`request get, no params 1`] = `undefined`;

exports[`request get, no params 2`] = `undefined`;

exports[`request list 1`] = `
Array [
  "/projects/projectId/merge_requests",
  Object {
    "simple": true,
  },
]
`;

exports[`request list 2`] = `
Array [
  Array [
    "/projects/projectId/merge_requests",
    Object {
      "simple": true,
    },
  ],
]
`;

exports[`request list with opts 1`] = `
Array [
  "/projects/projectId/merge_requests",
  Object {
    "simple": false,
  },
]
`;

exports[`request list with opts 2`] = `
Array [
  Array [
    "/projects/projectId/merge_requests",
    Object {
      "simple": false,
    },
  ],
]
`;

exports[`request list, no params 1`] = `undefined`;

exports[`request notes 1`] = `
Array [
  "/projects/projectId/merge_requests/mergeRequestId/notes",
  Object {
    "simple": true,
  },
]
`;

exports[`request notes 2`] = `
Array [
  Array [
    "/projects/projectId/merge_requests/mergeRequestId/notes",
    Object {
      "simple": true,
    },
  ],
]
`;

exports[`request notes, no params 1`] = `undefined`;

exports[`request notes, no params 2`] = `undefined`;

exports[`request unaprove 1`] = `undefined`;
